File: matiasmasso/Mat.NET/DT2/DTOBaseId.vb
Public Class DTOBaseId
    Private _Id As Integer

    Property AuxObject As Object

    Property IsNew As Boolean
    Property IsLoaded As Boolean

    Public Sub New(iId As Integer)
        MyBase.New()
        _Id = iId
    End Sub

    ReadOnly Property Id As Integer
        Get
            Return _Id
        End Get
    End Property

    Public Shadows Function UnEquals(oCandidate As DTOBaseId) As Boolean
        Dim retval As Boolean = Not Equals(oCandidate)
        Return retval
    End Function

    Public Shadows Function Equals(oCandidate As DTOBaseId) As Boolean
        Dim retval As Boolean
        If oCandidate IsNot Nothing Then
            retval = _Id = oCandidate.Id
        End If
        Return retval
    End Function

End Class

File: matiasmasso/Mat.NET/DTO/DTOBaseId.cs
namespace DTO
{
    public class DTOBaseId
    {
        private int _Id;

        public object AuxObject { get; set; }

        public bool IsNew { get; set; }
        public bool IsLoaded { get; set; }

        public DTOBaseId(int iId) : base()
        {
            _Id = iId;
        }

        public int Id
        {
            get
            {
                return _Id;
            }
        }

        public bool UnEquals(DTOBaseId oCandidate)
        {
            bool retval = !Equals(oCandidate);
            return retval;
        }

        public bool Equals(DTOBaseId oCandidate)
        {
            bool retval = false;
            if (oCandidate != null)
                retval = _Id == oCandidate.Id;
            return retval;
        }
    }
}
